[terminal] link matcher doesn't work with rust / go cli output

Rust and go use the following scheme:

relative/path:start:end:

which is not recognized.

Example for go:
```
# github.com/demo-apps/go-gin-app
./models.article.go:16:11: undefined: leList
```

Example for Rust:
```
error[E0061]: this function takes 1 parameter but 2 parameters were supplied
  --> src/app.rs:45:10
   |
45 |         .arg(
   |          ^^^ expected 1 parameter

error[E0599]: no function or associated item named `with` found for type `clap::Arg<'_, '_>` in the current scope
  --> src/app.rs:46:18
   |
46 |             Arg::with _name("case-sensitive")
   |                  ^^^^ function or associated item not found in `clap::Arg<'_, '_>`

```

--- a/packages/process/src/node/process.ts
+++ b/packages/process/src/node/process.ts
@@ -196,10 +196,6 @@ export abstract class Process {
             });
         } else if (!isWindows) {
             return new Promise<string>(resolve => {
-                resolve(FileUri.create(this.initialCwd).toString());
-            });
-        } else {
-            return new Promise<string>(resolve => {
                 fs.readlink('/proc/' + this.pid + '/cwd', (err, linkedstr) => {
                     if (err || !linkedstr) {
                         resolve(FileUri.create(this.initialCwd).toString());
@@ -207,6 +203,10 @@ export abstract class Process {
                         resolve(FileUri.create(linkedstr).toString());
                     }
                 });
+            });
+        } else {
+            return new Promise<string>(resolve => {
+                resolve(FileUri.create(this.initialCwd).toString());
             });
         }
     }

--- a/packages/terminal/src/browser/base/terminal-service.ts
+++ b/packages/terminal/src/browser/base/terminal-service.ts
@@ -40,6 +40,10 @@ export interface TerminalService {
 
     readonly all: TerminalWidget[];
 
+    /**
+     * @param id - the widget id (NOT the terminal id!)
+     * @return the widget
+     */
     getById(id: string): TerminalWidget | undefined;
 
     readonly onDidCreateTerminal: Event<TerminalWidget>;

--- a/packages/terminal/src/browser/terminal-widget-impl.ts
+++ b/packages/terminal/src/browser/terminal-widget-impl.ts
@@ -207,7 +207,7 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
         if (!IBaseTerminalServer.validateId(this.terminalId)) {
             throw new Error('terminal is not started');
         }
-        if (this.terminalService.getById(this.terminalId.toString())) {
+        if (this.terminalService.getById(this.id)) {
             return this.shellTerminalServer.getCwdURI(this.terminalId)
                 .then(cwdUrl => new URI(cwdUrl));
         }
@@ -469,7 +469,7 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
 
     protected resizeTerminalProcess(): void {
         if (!IBaseTerminalServer.validateId(this.terminalId)
-            && !this.terminalService.getById(this.terminalId.toString())) {
+            && !this.terminalService.getById(this.id)) {
             return;
         }
         const { cols, rows } = this.term;

